docs/asciidoc/debugging.adoc

```diff
@@ -319,6 +323,74 @@ NOTE: When both global debugging and local `checkpoint()` are enabled, checkpoin
 snapshot stacks are appended as suppressed error output after the observing operator
 graph and following the same declarative order.
 
+[[reactor-tools-debug]]
+== Production-ready debugging
+Project Reactor comes with a separate Java Agent that instruments your code and adds
+debugging info without paying the cost of capturing the stacktrace on every operator call.
+The behaviour is very similar to <<debug-activate>>, but without the runtime performance overhead.
+
+To use it in your app, you must add it as a dependency.
+
+The following example shows how to add `reactor-tools` as a dependency in Maven:
+
+.reactor-tools in Maven, in `<dependencies>`
+====
+[source,xml]
+----
+<dependency>
+    <groupId>io.projectreactor</groupId>
+    <artifactId>reactor-tools</artifactId>
+    <1>
+</dependency>
+----
+<1> If you use the <<getting,BOM>>, you do not need to specify a `<version>`.
+====
+
+The following example shows how to add `reactor-tools` as a dependency in Gradle:
+
+.reactor-tools in Gradle, amend the `dependencies` block
+====
+[source,groovy]
+----
+dependencies {
+   compile 'io.projectreactor:reactor-tools'
+}
+----
+====
+
+It also needs to be explicitly initialized with:
+====
+[source,java]
+----
+ReactorDebugAgent.init();
+----
+====
+
+TIP: Since the implementation will instrument your classes when they are loaded,
+the best place to put it is before everything else in your main(String[]) methood:
+====
+[source,java]
+----
+public static void main(String[] args) {
+    ReactorDebugAgent.init();
+    SpringApplication.run(Application.class, args);
+}
+----
+====
+
+You may also re-process existing classes if you cannot run the init eagerly (e.g. in the tests):
+====
+[source,java]
+----
+ReactorDebugAgent.init();
+ReactorDebugAgent.processExistingClasses();
+----
+====
+
+WARNING: Be aware that the re-processing takes a couple of seconds due to the need to iterate over
+all loaded classes and apply the transformation.
+Use it only if you see that some call-sites are not instrumented.
+
```

Comment:
also add a note about how programmatic-style of installing agent is not guaranteed to work on all JDK flavors / JREs